nautobot/core/views/mixins.py

```diff
@@ -217,8 +217,50 @@ def get_return_url(self, request, obj=None, default_return_url=None):
         return reverse("home")
 
 
+ComponentType = TypeVar("ComponentType")
+
+
+class ComponentsMixin:
```

Comment:
1) Please add an appropriate docstring, explaining usage and when/where this mixin should be used.
2) If apps can/should/must use this mixin, please be sure it's exposed in `nautobot.apps` submodule namespace.
3) Can we find a less generic name? Thinking particular about the ambiguity with *device* components like Interface. Maybe `BreadcrumbsTitlesMixin`, `UIComponentsMixin`, or something like that?